src/pu-emmc.c

```diff
@@ -831,6 +862,55 @@ pu_emmc_parse_partitions(PuEmmc *emmc,
     return TRUE;
 }
 
+static gboolean
+pu_emmc_check_raw_overwrite(PuEmmc *emmc,
```

Comment:
Do you like to add a unit test for this function? I think adding different layout configs with non-overlapping and overlapping raw binaries could be used to test this.

Reply:
Added 4 test for the different overwrite scenarios.